packages/utils/src/create-extended-api.ts

```diff
@@ -86,6 +86,13 @@ export function createExtendedSyncActions(baseFs: IBaseFileSystemSync): IFileSys
       if (parentPath === directoryPath) {
         throw e;
       }
+
+      // Windows also throws ENOENT when trying to create a directory inside of a file,
+      // unlike Mac/Linux that throw ENOTDIR.
+      if (fileExistsSync(parentPath)) {
```

Comment:
why is this extra check needed? the recursive call to `ensureDirectorySync(parentPath);` should be the one throwing that exception when it tries to `mkdirSync` that directory.

also, it's important than any change to the sync variants is also applied to the async variants (also in `create-extended-api.ts`)

Reply:
Scenario: creating a directory inside of a file.

After `mkdirSync('/foo.txt/dir')` Mac and Linux throw ENOTDIR, and we're done. But Windows throws ENOENT (same as when the parent directory doesn't exist). Without checking `fileExistsSync(parentPath)` we would recurse and try to create the parent directory `/foo.txt`, and would get `EEXIST`, which is quite misleading, as if `/foo.txt/dir` already exists.

Reply:
I've merged #104 so that paths are set on the Error objects themselves.
In the case of `ensureDirectory`, when it re-throws the recursive call's exception, that error will contain the actual path that failed.

Given that this function will soon be a minor wrapper to `mkdir(p, {recursive: true})`, I wouldn't worry too much about the error codes for the edge cases. Instead I rather it be optimized for the common case, and later let the error codes from node's native recursive implementation to propagate. see #103 for script + output on Linux/Windows.